dvc/utils/compat.py

```diff
@@ -85,8 +85,6 @@ def _makedirs(name, mode=0o777, exist_ok=False):
             if e.errno != errno.EEXIST:
                 raise
         cdir = os.curdir
-        if isinstance(tail, bytes):
```

Comment:
Well, it will sound picky:
what if someone will try to `_makedirs(b'/tmp/test/something/.')` in python3? 
According to little test I did, we will get FileExistError, because this check will be skipped. 
We can either accept this situation, or check if its py_3 before that one.

Reply:
@pared that was the problem of this code - it runs exclusively on py2 as far as I udnerstand, but `str` function call (and byte type check) were taken from Python 3 implementation which actually breaks on python 2

Reply:
It couldn't run only on py2, as behavior for py3 is also changed. When we run my example with this change we immediately get `FileExistError`, while without it, the code workes without exception. 
here is the script:
```import os, shutil
from dvc.utils.compat import _makedirs

if os.path.exists("/tmp/test"):
    shutil.rmtree("/tmp/test")

_makedirs(b'/tmp/test/something/.')```

Reply:
so, we never use `from dvc.utils.compat import _makedirs`, we do instead `from dvc.utils.compat import makedirs` (mind the `_` prefix difference). `_makedirs` is an internal package member, and actual `compat.makedirs`  is defined below - lines 123 and 147 (this branch).

Reply:
Ahh got it, yes, you are right, sorry, that was unnecessary.

Reply:
👍 thanks for reviewing this @pared ! There was a bug actually (not just extra non-used code) - those tests on Py2 you told that are failing were due to this bug.